### ISSUE TITLE
ClientSecretCredential.get_token() fails in case for azurestack if we provide 'authority' URL 

### ISSUE THREAD

**Issue by Mrbm5** [Azure/azure-sdk-for-python#22625]
- **Package Name: azure-identity**: 
- **Package Version: 1.6.1**: 
- **Operating System: ubuntu20.04**:
- **Python Version: 3.8.10**: 

**Describe the bug**
To authenticate python SDK to azure Stack hub we use ClientSecretCredential method and pass required parameters like 
`ClientSecretCredential(
            tenant_id=tenantid, client_id=clientid,
            client_secret=secretkey, cloud_environment=cloud,
            authority=auth_resource).`

when user pass authority url '**https://management.veritasazs.onmicrosoft.com/<b9ae38c5-####>**' parameter expilicitly the ClientSecretCredential.get_token() method throwing an error instead of successful response. 
ClientSecretCredential.get_token failed: Authentication failed: invalid_instance: The authority you provided, https://management.veritasazs.onmicrosoft.com/<b9ae38c5-####>/<ed83b0eb-tenant_id>, is not whitelisted. If it is indeed your legit customized domain name, you can turn off this check by passing in validate_authority=False. 

looks like the azure sdk explicitly appending tenant_id to the authority URL. 
can you please provide us more information about the change and how user can pass the authority URL.
if we dont pass 'authority URL' then its working but with authority URL it fails. 

**To Reproduce**
1. try to provide authority url along with other required parameters while creating ClientSecretCredential. 

**Expected behavior**
ClientSecretCredential.get_token() should work if user pass the authority URL.

**Comment by xiangyan99** [comment]
@rayluo I see the issue raises from msal.Authority. 

Could you take a look?

**Comment by rayluo** [comment]
> @rayluo I see the issue raises from msal.Authority.
> 
> Could you take a look?

Yes, that exception was thrown from [here](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/1.16.0/msal/authority.py#L62-L73). The solution, as hinted by the exception message, "If it is indeed your legit customized domain name, you can turn off this check by passing in validate_authority=False." That would mean, MSAL Python's downstream app would need to utilize [this parameter](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/1.16.0/msal/application.py#L236-L237). If Azure Identity somehow feed that `validate_authority` flag with proper value, MSAL Python would accept the authority. Does Azure Identity expose that optional parameter to downstream apps?

**Comment by xiangyan99** [comment]
@rayluo Thanks for the information. Unfortunately, we don't have validate_authority support yet.

I created PR #22786 to support it.

**Comment by xiangyan99** [comment]
@Mrbm5 if you don't mind, could you share more information with us?

e.g. are you using ADFS? I ask this because https://management.private_domain.onmicrosoft.com/<b9ae38c5-####>/<ed83b0eb-tenant_id> seems not a valid authority url.

If you are sure this should work, could you try disable validate_authority by using the work around in #22786?

Thank you.

**Comment by Mrbm5** [comment]
@xiangyan99 No, Im using Azure Active Directory(AAD) for authentication. and the original authority URL I'm passing is : : **https://management.itasxxxx.onmicrosoft.com/b9ae38c5-xxxx-xxxx-xxxx-xxxxxxxxc6**. we have used this URL successfully with `ServicePrincipalCredentials`. with the recent SDK update to `azure-identity==1.6.1` using `ClientSecretCredential` and now for same authority url we are getting this error ClientSecretCredential.get_token failed: Authentication failed: invalid_instance: The authority you provided, **https://management.itasxxxx.onmicrosoft.com/b9ae38c5-xxxx-xxxx-xxxx-xxxxxxxxc6/ed83b0eb-xxxx-xxxx-xxxx-xxxxxxxxb6**, is not whitelisted. If it is indeed your legit customized domain name, you can turn off this check by passing in validate_authority=False.

looking at this URL looks like the tenant_id is getting added  to original url.  but the authority url provided by us is already a valid url. IMO we don't need to append tenant_id to it. 

if we dont pass any authority url it takes 'https://login.microsoftonline.com/ed83b0eb-xxxx-xxxx-xxxx-xxxxxxxxb6', in this case this is the valid one. 

Hope this will help to understand the problem statement. Please le me know if you need more information.

**Comment by xiangyan99** [comment]
@Mrbm5 Thank you for providing more information. This really helps us understand the scenario. In #22786 we will expose `validate_authority` parameter and you can explicitly disable the validation.

**Comment by Mrbm5** [comment]
Thanks @xiangyan99. can you please tell us in which version this change will be available or when it will be available to use?

**Comment by xiangyan99** [comment]
We plan to release it next week with version number 1.8.0b1.

**Comment by bganapa** [comment]
@Mrbm5  You cannot use the Token Audience https://management.itasxxxx.onmicrosoft.com/b9ae38c5-xxxx-xxxx-xxxx-xxxxxxxxc6 as Authority URI, The Authority URI has to be 'https://login.microsoftonline.com', The tenant id ed83b0eb-xxxx-xxxx-xxxx-xxxxxxxxb6 will get appended while requesting for the token.

pls refer to our azure stack samples here https://github.com/Azure-Samples/Hybrid-Python-Samples/blob/4c3f4d96e18e5af575cf395830e53f5fed860516/resourcemanager/example.py#L37

The following line reads the authority uri from the ARM metadata endpoint
https://github.com/Azure-Samples/Hybrid-Python-Samples/blob/4c3f4d96e18e5af575cf395830e53f5fed860516/resourcemanager/example.py#L30

also pls note the scope value passed in the following ine
https://github.com/Azure-Samples/Hybrid-Python-Samples/blob/4c3f4d96e18e5af575cf395830e53f5fed860516/resourcemanager/example.py#L45


@keystroke FYI

**Comment by xiangyan99** [comment]
The new library is released.

**Comment by keystroke** [comment]
@xiangyan99 thanks for making the validate authority fix, that is a good change and authority validation can be disabled in almost all cases.

However as @bganapa indicated in above comment, this is actually a user input error and disabling validate authority will not resolve the issue.

**Comment by rayluo** [comment]
> authority validation can be disabled in almost all cases.

To clarify, we recommend ENABLE authority validation in almost all cases. The reason was hinted in the MSAL Python's error message, "*If it is indeed your legit customized domain name*, you can turn off this check by passing in `validate_authority=False`." For example, if the app developer is developing a Line-Of-Business (LOB) app for his employer only, that app could hardcode their customized domain "`https://contoso.com/adfs`" and then turn off authority validation. Do not implement an app that would dynamically load authority such as "`https://i.steal.your.password.com`" from somewhere else, and opt out of authority validation.

**Comment by anotherancientalien** [comment]
> Azure Identity January - April 2022 Releases Azure/azure-sdk-for-go#16728

validate_authority=False is not working. The problem still exist and all our codes for RSV are failed. Please provide solution.
